src/plugins/intel_gpu/src/kernel_selector/kernel_selector_common.h

```diff
@@ -114,6 +114,21 @@ struct KernelData {
 
     bool can_reuse_memory = true;
 
+    template <typename op_params>
+    static bool SkipKernelExecution(const op_params& params, size_t kernel_id = 0) {
```

Comment:
It is possible, but then instead of `KernelData::SkipKernelExecution(prim_params);` it will be `KernelData::SkipKernelExecution(dynamic_cast<const base_params&>(prim_params));`
Just shortened it a bit

Reply:
dynamic cast is not needed as it's just upcasting to base class ref.
Extensive templates usage impacts lib binary size, so I think in that case single (even non-virtual) function is better